internal/app/upstream/client.go

```diff
@@ -360,3 +367,19 @@ func updateConnectivityMetric(ctx context.Context, conn *grpc.ClientConn, scope
 		scope.Gauge(metrics.UpstreamConnected).Update(float64(conn.GetState()))
 	}
 }
+
+func getKeepaliveParams(ctx context.Context, logger log.Logger, c CallOptions) keepalive.ClientParameters {
+	keepaliveClientParams := keepalive.ClientParameters{
+		PermitWithoutStream: true,
```

Comment:
do we need to set this?

Reply:
`PermitWithoutStream` allows keepalives to be sent even if there are no streams. This will keep the cx warm and have no cx establishment penalty on the first stream. grpc closes the cx if there are no activew streams and this setting will prohibit that.

Reply:
The `ClientParameters` already has the documentation, so i did not repeat them here